### ISSUE TITLE
expect_column_pair_values_a_to_be_greater_than_b fail to show unexpected rows

### ISSUE THREAD

**Issue by VytSep** [great-expectations/great_expectations#10323]
**Describe the bug**
The expectation expect_column_pair_values_a_to_be_greater_than_b is not generating the unexpected rows

**To Reproduce**
The expectation is configured this way:
COMPLETE={"result_format": "COMPLETE", "include_unexpected_rows": True}
expectation = gx.expectations.ExpectColumnPairValuesAToBeGreaterThanB(
    column_A="date_a",
    column_B="date_b",
    or_equal=True,
    result_format=COMPLETE,
)

The checkpoint is configured this way:
checkpoint_name = "my_checkpoint"
checkpoint = gx.Checkpoint(
    name=checkpoint_name,
    validation_definitions=validation_definitions,
    result_format={"result_format": "COMPLETE", "include_unexpected_rows": True},
)

**Expected behavior**
Have in the validation result the unexpected rows for this expectation

**Environment (please complete the following information):**
 - Great Expectations Version: [1.0.0]
 - Cloud environment: [Databricks]

**Additional context**
For other expectations like expect_column_values_to_be_in_set I do have the unexpected rows generated

**Comment by adeola-ak** [comment]
hi there - just want to let you know i am looking into this. to my knowledge, both should be returning unexpected rows. Please stand by while I continue to investigate

**Comment by adeola-ak** [comment]
following up here, what are you getting it back? do you see the key returning the object for unexpected rows like you do for expect_column_values_to_be_in_set at all?

**Comment by VytSep** [comment]
I do have key returning for the expect_column_values_to_be_in_set and also for the expect_column_pair_values_a_to_be_greater_than_b

**Comment by adeola-ak** [comment]
so you have the unexpected values key being returned for both but for `expect_column_pair_values_a_to_be_greater_than_b` the key is just empty? can you share a ss?

**Comment by VytSep** [comment]
Yes, I do have the keys for both expectations. What do you mean for ss?

**Comment by adeola-ak** [comment]
screenshot i'm sorry

**Comment by VytSep** [comment]
Sorry, I don't have permission to share screenshots

**Comment by adeola-ak** [comment]
understood no problem! but again, is the `unexpected_rows` key empty or returning null or something else?

if you edit your project to only have the expect_column_pair_values_a_to_be_greater_than_b  expectation, and run `print(validation_results["results"][0]["result"]["unexpected_rows"])
` what do you see? As i'm seeing None across a few projects, I do need to clarify again what is expected with such an expecation that is evaluated for a pair of columns. I will get a concrete answer to you by today

**Comment by VytSep** [comment]
![Screenshot 2024-09-19 120452](https://github.com/user-attachments/assets/774ce7a6-bbdd-4c3d-ab2a-1268e3d08c00)

**Comment by VytSep** [comment]
Hi team, any news on that subject?

**Comment by adeola-ak** [comment]
Hi @VytSep, thank you for your patience. This issue has been confirmed as a bug. Unfortunately, I don't have a specific timeline for when it will be addressed, but I plan to follow up with the team tomorrow and should have an update then.

**Comment by adeola-ak** [comment]
@VytSep did this work for you in the past? or prior to 1.0?

**Comment by VytSep** [comment]
Hi @adeola-ak , no i did not. I have the same issue also with the 0.18.15 version.

**Comment by adeola-ak** [comment]
hi @VytSep sorry that I misspoke earlier this month. After further review, though unexpected rows works for `ColumnMapExpectations`, this behavior is expected for all `ColumnPairMapExpectations`. I will update our documentation to reflect this. For now, having this functionality would be considered a feature request, which is not prioritized at the moment. Thank you for your patience, and I apologize if this isn't the outcome you were hoping for.

**Comment by wookasz** [comment]
As part of triaging open issues, I have confirmed that unexpected rows are now returned for this Expectation. I'm closing this issue. Feel free to re-open if this is still not working for you.
